executor/infoschema_reader.go

```diff
@@ -1779,6 +1782,51 @@ func (e *memtableRetriever) setDataForStatementsSummary(ctx sessionctx.Context,
 	return nil
 }
 
+func (e *memtableRetriever) setDataForPlacementPolicy(ctx sessionctx.Context) error {
+	checker := privilege.GetPrivilegeManager(ctx)
+	is := infoschema.GetInfoSchema(ctx)
+	ruleBundles := is.RuleBundles()
+	var rows [][]types.Datum
+	for _, bundle := range ruleBundles {
+		id, err := placement.ObjectIDFromGroupID(bundle.ID)
+		if err != nil {
+			return errors.Wrapf(err, "Restore bundle %s failed", bundle.ID)
+		}
+		if id == 0 {
+			continue
+		}
+		// Currently, only partitions have placement rules.
+		tb, db, part := is.FindTableByPartitionID(id)
+		if tb == nil {
+			return errors.Errorf("Can't find partition by id %d", id)
+		}
+		if checker != nil && !checker.RequestVerification(ctx.GetSessionVars().ActiveRoles, db.Name.L, tb.Meta().Name.L, "", mysql.SelectPriv) {
+			continue
+		}
+		for _, rule := range bundle.Rules {
+			constraint, err := placement.RestoreLabelConstraintList(rule.LabelConstraints)
+			if err != nil {
+				return errors.Wrapf(err, "Restore rule %s in bundle %s failed", rule.ID, bundle.ID)
+			}
+			row := types.MakeDatums(
+				bundle.ID,
+				bundle.Index,
+				rule.ID,
+				db.Name.L,
+				tb.Meta().Name.L,
+				part.Name.L,
+				nil,
+				string(rule.Role),
+				rule.Count,
+				constraint,
+			)
+			rows = append(rows, row)
+		}
+	}
+	e.rows = rows
+	return nil
```

Comment:
```suggestion
```

Reply:
It can't be removed, the function needs a return value.